linkerd/app/outbound/src/http/require_id_header.rs

```diff
@@ -65,32 +69,41 @@ where
         self.inner.poll_ready(cx).map_err(Into::into)
     }
 
-    fn call(&mut self, request: http::Request<A>) -> Self::Future {
-        // If the `l5d-require-id` header is present, then we should expect
-        // the target's `peer_identity` to match; if the two values do not
-        // match or there is no `peer_identity`, then we fail the request
-        if let Some(require_id) = identity_from_header(&request, L5D_REQUIRE_ID) {
-            debug!("found l5d-require-id={:?}", require_id.as_ref());
+    fn call(&mut self, mut request: http::Request<B>) -> Self::Future {
+        // If the `l5d-require-id` header is present, then we should expect the target's
+        // `peer_identity` to match; if the two values do not match or there is no `peer_identity`,
+        // then we fail the request.
+        //
+        // In either case, we clear the header so it is not passed on outbound requests.
+        if let Some(require_id) = Self::extract_id(&mut request) {
             match self.tls.as_ref() {
                 Conditional::Some(tls::ClientTls { server_id, .. }) => {
                     if require_id != *server_id.as_ref() {
+                        debug!(
+                            required = %require_id,
+                            found = %server_id,
+                            "Identity required by header not satisfied"
+                        );
                         let e = IdentityRequired {
                             required: require_id.into(),
                             found: Some(server_id.clone()),
                         };
-                        return Either::Left(future::err(e.into()));
+                        return future::Either::Left(future::err(e.into()));
+                    } else {
+                        trace!(id = %require_id, "Identity required by header");
                     }
                 }
                 Conditional::None(_) => {
+                    debug!(id = %require_id, "Identity required by header not satisfied");
```

Comment:
nit: should these be
```suggestion
                    } else {
                        trace!(required = %require_id, "Identity required by header");
                    }
                }
                Conditional::None(_) => {
                    debug!(required = %require_id, "Identity required by header not satisfied");
```
for consistency with the other debug event?